docs/DatabaseInjector.md

```diff
@@ -0,0 +1,106 @@
+# Database Injector Guide
+
+NuoDB Control Plane (CP) can be used to create NuoDB databases for applications that are running in the same [Kubernetes][1] cluster.
+This document describes how to use database injector to supply database connection information to sample YCSB application.
+
+## Prerequisites
+
+- A running [Kubernetes cluster][2]
+- [kubectl][3] installed and able to access the cluster.
+
+## Installing NuoDB Control Plane
+
+Install NuoDB CP as documented in [Installation](../README.md#installation) section.
+
+### Enable Cluster-scoped Access
+
+By default NuoDB operator will monitor only the local namespace for NuoDB [custom resources][4].
+If cluster-scoped access is required, set the `cpOperator.watchNamespace.enabled=false` Helm value during installation. E.g.:
+
+```sh
+helm install nuodb-cp-operator nuodb-cp/nuodb-cp-operator \
+    --namespace nuodb-cp-system \
+    --set cpOperator.watchNamespace.enabled=false \
+    --set nuodb.serviceAccount.create=false \
+    --set nuodb.serviceAccount.name=default \
+    ...
+```
+
+>**Note**
+> The `nuodb` service account (SA) creation is disabled in the above command for simplicity. To enable NuoDB Kubernetes Aware Admin (KAA) capabilities, the NuoDB processes should be given special permissions to access Kubernetes API server.
+For more information, please check [Automatic Management of NuoDB State](https://doc.nuodb.com/nuodb/latest/deployment-models/kubernetes-environments/kubernetes-aware-admin/). For cluster-scoped deployments, the NuoDB SA and RBAC should be provisioned before hand in each namespace where NuoDB databases are created.
+
+## Database Injection
+
+NuoDB operator can inject database information into ConfigMap's _data_ once the database is ready.
+This enables easy data source configuration in the application container and acts as a dependency mechanism without the need of additional _init_ containers.
```

Comment:
This isn't so clear without more context or an example. Can you say that the database connection details are populated _after_ the database is created, and this blocks scheduling of the application pods due to the configmap value being absent. Maybe some example output with the pod in the "ErrorConfig" or whatever it's called, would be useful.

Reply:
Will do.